src/Diagnostics.AIProjects/SearchAPI/SearchModule/TextSearchModule.py

```diff
@@ -23,7 +38,14 @@ def __init__(self, modelpackagepath):
             self.model = WmdSearchModel(modelpackagepath)
 
     def queryDetectors(self, query=None):
-        return self.model.queryDetectors(query)
+        cleansed_query = " ".join(re.sub(r'[^(0-9a-zA-Z )]+', " ", query).split())
+        # Break the query into smaller chunks if number of word is greater than 6
+        if len(cleansed_query.split())>6:
```

Comment:
So this means that the original query has more than 6 words?

Reply:
Yup, if the original query has more than 6 words, we break it down.